types/dragino/lht65n-ne117/uplink.js

```diff
@@ -0,0 +1,277 @@
+function Str1(str2) {
+  let str3 = "";
+  for (let i = 0; i < str2.length; i++) {
+    if (str2[i] <= 0x0f) {
+      str2[i] = `0${str2[i].toString(16)}`;
+    }
+    str3 += `${str2[i].toString(16)}`;
+  }
+  return str3;
+}
+
+function strPad(byte) {
+  const zero = "00";
+  const hex = byte.toString(16);
+  const tmp = 2 - hex.length;
+  return `${zero.substr(0, tmp) + hex} `;
+}
+
+function getzf(cNum) {
+  if (parseInt(cNum) < 10) {
+    cNum = `0${cNum}`;
+  }
+
+  return cNum;
+}
+
+function getMyDate(str) {
+  let cDate;
+  if (str > 9999999999) {
+    cDate = new Date(parseInt(str));
+  } else {
+    cDate = new Date(parseInt(str) * 1000);
+  }
+
+  const cYear = cDate.getFullYear();
+  const cMonth = cDate.getMonth() + 1;
+  const cDay = cDate.getDate();
+  const cHour = cDate.getHours();
+  const cMin = cDate.getMinutes();
+  const cSen = cDate.getSeconds();
+  const cTime = `${cYear}-${getzf(cMonth)}-${getzf(cDay)} ${getzf(
+    cHour,
+  )}:${getzf(cMin)}:${getzf(cSen)}`;
+
+  return cTime;
+}
+
+function datalog(i, bytes) {
+  const Ext = bytes[6] & 0x0f;
+  let bb;
+  if (Ext === "1" || Ext === "9") {
+    bb = parseFloat(
+      ((((bytes[0 + i] << 24) >> 16) | bytes[1 + i]) / 100).toFixed(2),
+    );
+  } else if (Ext === "2") {
+    bb = parseFloat(
+      ((((bytes[0 + i] << 24) >> 16) | bytes[1 + i]) / 100).toFixed(2),
+    );
+  } else if (Ext === "4") {
+    const extiPinLevel = bytes[0 + i] ? "HIGH" : "LOW";
+    const extiStatus = !!bytes[1 + i];
+    bb = extiPinLevel + extiStatus;
+  } else if (Ext === "5") {
+    bb = (bytes[0 + i] << 8) | bytes[1 + i];
+  } else if (Ext === "6") {
+    bb = ((bytes[0 + i] << 8) | bytes[1 + i]) / 1000;
+  } else if (Ext === "7") {
+    bb = (bytes[0 + i] << 8) | bytes[1 + i];
+  } else if (Ext === "8") {
+    bb = (bytes[0 + i] << 8) | bytes[1 + i];
+  }
+  const cc = parseFloat(
+    ((((bytes[2 + i] << 24) >> 16) | bytes[3 + i]) / 100).toFixed(2),
+  );
+  const dd = parseFloat(
+    ((((bytes[4 + i] << 8) | bytes[5 + i]) & 0xfff) / 10).toFixed(1),
+  );
+  const ee = getMyDate(
+    (
+      (bytes[7 + i] << 24) |
+      (bytes[8 + i] << 16) |
+      (bytes[9 + i] << 8) |
+      bytes[10 + i]
+    ).toString(10),
+  );
+  const string = `[${bb},${cc},${dd},${ee}]` + `,`;
+
+  return string;
+}
+
+function decoder(bytes, port) {
+  const ext = bytes[6] & 0x0f;
+  const pollMessageStatus = (bytes[6] >> 7) & 0x01;
+  const connect = (bytes[6] & 0x80) >> 7;
+  const decode = { lifecycle: {}, decoded: {}, external: {}, datalog: {} };
+  if (((port === 3) & (bytes[2] === 0x01)) | (bytes[2] === 0x02)) {
+    const array1 = [];
```

Comment:
try to refactor this method to use more speaking names, if it was provided like this from the manufacturer it is fine and you can leave it